cudax/include/cuda/experimental/__utility/driver_api.cuh

```diff
@@ -0,0 +1,84 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of CUDA Experimental in CUDA C++ Core Libraries,
+// under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _CUDAX__UTILITY_DRIVER_API
+#define _CUDAX__UTILITY_DRIVER_API
+
+#include <cuda.h>
+
+#include <cuda/std/__exception/cuda_error.h>
+
+// Get the driver function by name using this macro
+#define CUDAX_GET_DRIVER_FUNCTION(function_name) \
+  reinterpret_cast<decltype(function_name)*>(get_driver_entry_point(#function_name))
+
+namespace cuda::experimental::detail::driver
+{
+inline void* get_driver_entry_point(const char* name)
+{
+  void* fn;
+  cudaDriverEntryPointQueryResult result;
+  cudaGetDriverEntryPoint(name, &fn, cudaEnableDefault, &result);
+  if (result != cudaDriverEntryPointSuccess)
+  {
+    if (result == cudaDriverEntryPointVersionNotSufficent)
+    {
+      ::cuda::__throw_cuda_error(cudaErrorNotSupported, "Driver does not support this API");
+    }
+    else
+    {
+      ::cuda::__throw_cuda_error(cudaErrorUnknown, "Failed to access driver API");
+    }
+  }
+  return fn;
+}
+
+template <typename Fn, typename... Args>
+inline void call_driver_fn(Fn fn, const char* err_msg, Args... args)
+{
+  CUresult status = fn(args...);
+  if (status != CUDA_SUCCESS)
+  {
+    ::cuda::__throw_cuda_error(static_cast<cudaError_t>(status), err_msg);
+  }
```

Comment:
It could also be a function.

Reply:
This function should be more or less equivalent to _CCCL_TRY_CUDA_API, am I missing some key difference here? I would have no issues turning it into a macro instead if its preffered

Reply:
We might need two separate functions/macros because driver API returns CUresult and runtime returns cudaError_t.

But these have the same values, so maybe we can add a cast to _CCCL_TRY_CUDA_API and remove this function 🤔